### ISSUE TITLE
Inspect purpose of web paths that cannot be viewed and produces internal server error

### ISSUE THREAD

**Issue by carlosparadis** [erdl/survey_admin#2]
* /questionform					
    * Template: question_form.html
  * /showdevices
    * Template: show_devices.html

We should decide what to do with them afterward once the purpose is clear.

**Comment by jygh98** [comment]
## /questionform

when trying to access the web path, it leads to an error page of

     NameError: name 'QuestionForm' is not defined

After checking forms.py, it appears that class QuesionForm was triple quoted.  

      '''
          class QuestionForm(FlaskForm):
      ''' 
After unignoring the class QuestionForm and reloading the page here is the resulting page

<img width="961" alt="questionform" src="https://user-images.githubusercontent.com/25787662/28599487-a6f64fa4-7146-11e7-848b-06cc01f50fea.png">

From what I recall of the video, Allie mentioned that previously we would add questions through the dashboard.  However, she decided to remove it in fear of the dashboard being broken into and people maliciously adding questions that was not meant to be there.  For that, any future questions to be added is done directly through the PostgreSQL server.  Also if you haven't notice there is no option to add questions on the dashboard as compared to adding surveys and deployments.

**Comment by carlosparadis** [comment]
When it is uncommented, are you able to add questions to the database? Give it a shot and see what happens. We should handle the code differently depending on it being working code or not.

**Comment by jygh98** [comment]
If I try to submit anything, it would result in an error

      TypeError: __init__() takes 1 positional argument but 3 were given

**Comment by carlosparadis** [comment]
Do you understand the cause of the error? Is this yet another uncommented part or something she didn't carry through implementing it? Try to make an assessment of the effort it would take for this to work, or conversely, how much of this (unwanted) "functionality" would take to be complete.

**Comment by jygh98** [comment]
The TypeError seems to be caused by an argument error which was corrected with [Celeo's response here](https://stackoverflow.com/questions/30032078/typeerror-init-takes-1-positional-argument-but-3-were-given).  However, a new error 

    sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) permission denied for sequence question_question_id_seq
    [SQL: 'INSERT INTO question (question_text, question_description, question_type) VALUES  
    (%(question_text)s, %(question_description)s, %(question_type)s) RETURNING question.question_id'] 
    [parameters: {'question_text': 'How tall are you\r\n', 'question_description': None, 'question_type': None}]

Currently looking into to what is causing this error.

**Comment by jygh98** [comment]
Error was solved with this [stackoverflow post](https://stackoverflow.com/questions/9325017/error-permission-denied-for-sequence-cities-id-seq-using-postgres)

**Comment by carlosparadis** [comment]
Right. We still need to understand what we are granting. There is a comment along the answer trail that hints at it:

> How is this a real thing? When would I ever want to allow a user to insert data into a table but NOT want to allow them to use the fact that one of the columns is auto-incrementing?

In fact, I am also curious about it. One other answer, by Asfand Qazi suggests that this will not apply to the rest of the tables and/or new table and something else must be done differently. 

TL;DR: We need to find an answer to the quoted question to make sure we are not granting more permissions than what we should to the app side (i.e. Allie's Survey Admin). 

Additionally, you forgot to mention that you **commented out** the `Description URL` from the code since the `Question` model on `models.py` did not contain any attribute associated with it. I imagine this is a legacy box when Survey_Admin was set to offer urls as well. This was deferred for Forrest app instead (This is pure guessing, you should check with @eileenpeppard) before it is also gone from her memory.

Don't forget we still have to look at `/showdevices` afterward for this issue.

**Comment by jygh98** [comment]
## /activedevices

When trying to access the webpath /activedevices, a nameError appears

    NameError: name 'ActiveQuestion' is not defined

After taking a look at the code, it makes sense as the class ActiveQuestion does not appear in either models.py and forms.py

    class ActiveQuestionForm(FlaskForm):
    urls=ActiveQuestion.query.with_entities(ActiveQuestion.activequestionurl)
    print("Hi")
    print(urls)
    activequestionurl = SelectField(u"activequestionurl", choices=[(q.activequestionid, q.activequestionurl) for q in ActiveQuestion.query.order_by('activequestionurl')])  

Both the ActiveQuestionForm and show_devices.html are completely written out, so in the event we do want a way to check active devices we will just need to write out the class ActiveQuestion.

**Comment by jygh98** [comment]
I looked around about the granting sequence permissions.  So from the [PostgreSQL 9.5 documentation](https://www.postgresql.org/docs/9.5/static/sql-grant.html), it states the following
>Granting permission on a table **does not** automatically extend permissions to any sequences used by the table, including sequences tied to SERIAL columns. Permissions on sequences must be set separately.

As of why this is a thing, [John Sidney-Woollett responded with the following](https://bytes.com/topic/postgresql/answers/400075-grant-question)
>I think you should provide an explicit grant (all) on the sequence as
well. The Sequence is an object that is distinct from the table that uses
it.

From what I read, it looks like granting select and usage permissions on sequence allows the use of the currval and nextval functions.

**Comment by carlosparadis** [comment]
@jygh98 I think the next step then is either see if roles from just a single database can be exported (so we can avoid granting more permissions than Allie did while avoiding getting all users from the entire database) and also seeing directly on the database how it was set. by her with `\d+ <table_name>` once you connect in with `psql`. Thoughts?

**Comment by jygh98** [comment]
@carlosparadis 
After reviewing the PR, it looked like we decided to completely remove show_devices and therefore removed it.

However i believed we simply commented out questionform in the event that we decide that we would like this functionality?  Questionform.html is still there along with /questionform in views.py and we did commented the [following
](https://github.com/erdl/survey_admin/blob/7f66d44887a41802f515581db65c536a067b9327/ubuntuStation/views.py#L105)
> disabled functionality to add question to the database (see github.com/erdl/survey_admin/ issue #2)

**Comment by carlosparadis** [comment]
OK we can revisit this down the road if we need to change anything!
